src/Microsoft.DotNet.Interactive/KeyValueStoreKernel.cs

```diff
@@ -26,6 +22,8 @@ public class KeyValueStoreKernel :
         internal const string DefaultKernelName = "value";
 
         private readonly ConcurrentDictionary<string, object> _values = new();
+        private ChooseKeyValueStoreKernelDirective _chooseKernelDirective;
+        private (bool hadValue, object previousValue, string newValue)? _lastOperation;
```

Comment:
Are these stored values? Why is `previousValue` an `object` while `newValue` is a `string`?

Reply:
type inference from refactoring tools, fixing that now